.github/workflows/publish_pypi.yml

```diff
@@ -27,22 +27,18 @@ jobs:
         with:
           python-version: 3.8
 
-      - name: Install cibuildwheel
-        run: python -m pip install cibuildwheel==2.12.3
-
       - name: Clone pybind11 repo (no history)
-        run: git clone --depth 1 --branch v2.10.4 https://github.com/pybind/pybind11.git
+        run: git clone --depth 1 --branch v2.11.1 https://github.com/pybind/pybind11.git
```

Comment:
It can be made a variable in a Lockfile of sorts or in the workflow file itself, but soon we should be able to remove this dependence on cloning a particular branch off the `pybind11` repository and let `pip` fetch the latest stable version. I have plans to do that after this is merged, as noted in [this comment](https://github.com/pybamm-team/PyBaMM/pull/3301?notification_referrer_id=NT_kwDOBG9FzrM3NTU3ODkzMjU2Ojc0NDAxMjMw#issuecomment-1731253161) which was lost in this super long thread haha